### PR TITLE
Update credential and cluster resource documentation for transparent proxy

### PR DESCRIPTION
**Pull request by shobha2626**
1. **What this PR does / why we need it**:

Update credential and cluster resource documentation for transparent proxy

3. **Which issue(s) this PR fixes**

        (optional, in `fixes #<issue number>` format, will close the issue(s) when PR gets merged):

        Fixes #

4. **Additional information**

5. **Special notes for your reviewer**

### CODE REVIEW

**Comment by Shubbhang351** [comment]
LGTM, but if required please do one round of reviews from service owner for docs.

File: docs/resources/credential.md
Comment (shobha2626) [reply docs/resources/credential.md:83]: We will merge this PR once the transparent proxy support is released.

File: examples/resources/credential/cluster_proxy.tf
Comment (shobha2626) [reply examples/resources/credential/cluster_proxy.tf:11]: If there is no annotation specified it will be considered as explicit proxy, for older proxy credential as well. There is no specific operation in such case

**Review by gouthamkn — APPROVED** [review]

**Review by ramya-bangera — CHANGES_REQUESTED** [review]
@shobha2626 do not edit the md files directly. Make the changes in the doc template and use `tfplugindocs`

File: examples/resources/credential/cluster_proxy.tf
Comment (vmw-vjn) [reply examples/resources/credential/cluster_proxy.tf:11]: @shobha2626 can we add a note in to [credential.md](https://github.com/vmware/terraform-provider-tanzu-mission-control/pull/219/files#diff-4b4bcc84a3b8fb567475215bbec3c11fde0580f3e7da460d1f0cb946e2eeb904) regarding proxy credentials created prior to v1.2.1

File: docs/data-sources/credential.md
Comment (vmw-vjn) [reply docs/data-sources/credential.md:40]: @shobha2626 - given that transparent proxy support in TMC is currently scheduled to go live on/by Aug 31. Amar is correct that documentation update regarding the same should possibly taken up in TMC TF provider's Sept 2023 release

cc: @ramya-bangera

File: examples/resources/credential/cluster_proxy.tf
Comment (shobha2626) [reply examples/resources/credential/cluster_proxy.tf:11]: @vmw-vjn I have captured it in here - https://github.com/vmware/terraform-provider-tanzu-mission-control/blob/f8bca04e22a74d384b1eca2d9894372021b7a659/docs/data-sources/credential.md?plain=1#L40

Comment (shobha2626) [reply examples/resources/credential/cluster_proxy.tf:11]: Removing this description as annotation is a common type used across resources.
Instead adding as a comment as suggested.

File: docs/resources/credential.md
Comment (ramya-bangera) [inline docs/resources/credential.md:75]: Is this added as part of template? I do not see that temple file changes here

Comment (shobha2626) [reply docs/resources/credential.md:75]: Missed it, added it now

File: examples/resources/credential/cluster_proxy.tf
Comment (ramya-bangera) [inline examples/resources/credential/cluster_proxy.tf:40]: nit: Please take care of the indentation throughout tf file.

**Review by ramya-bangera — APPROVED** [review]

File: docs/resources/cluster.md
Comment (shobha2626) [reply docs/resources/cluster.md]: Yes, moving image registry doc changes to separate PR

File: examples/resources/credential/cluster_proxy.tf
Comment (shobha2626) [reply examples/resources/credential/cluster_proxy.tf:40]: Done

**Review by vmw-vjn — COMMENTED** [review]
I see in credential.md as well as cluster_proxy.tf, the use of:
`"proxyCABundle" = "cabundle"`

Suggest adding comments alongside to specify the cert format & size limits (as applicable). As the same may not be evident for users, given that we have no data-source doc like equivalent for credentials which explains details of each variable used.

File: examples/resources/credential/cluster_proxy.tf
Comment (ramya-bangera) [inline examples/resources/credential/cluster_proxy.tf:28]: Have you validated this case just to ensure there are no issues with reading the certificates?

**Review by ramya-bangera — COMMENTED** [review]
Overall LGTM

File: examples/resources/credential/cluster_proxy.tf
Comment (shobha2626) [reply examples/resources/credential/cluster_proxy.tf:28]: Yes I have, the certificate is read correctly in this format

**Review by ramya-bangera — APPROVED** [review]

**Pull request merged**
